artemis-core-client/src/main/java/org/apache/activemq/artemis/core/client/impl/ClientSessionImpl.java

```diff
@@ -241,6 +243,16 @@
    // ClientSession implementation
    // -----------------------------------------------------------------
 
+   @Override
+   public boolean isEnable1xPrefixes() {
+      return enable1XPrefixes;
+   }
+
+   @Override
+   public void setEnable1xPrefixes(boolean value) {
```

Comment:
Does this need to be changable? Could this not just be set in constructor and field made final. As later the check is on hot path.

Reply:
@michaelandrepearce I needed it to be set because of the ResourceAdapter. MDBs.

Reply:
@michaelandrepearce Notice I'm improving the previous solution that had one boolean on each message.

Reply:
@michaelandrepearce also, notice this is only affecting the default on ActiveMQConnectionFactory...

There the value is also settable.

Reply:
Ive marked where hot paths concern me. On phone so may have missed one. Wont get to computer to properly go ovee till tomorrow

Reply:
Also all this 1x stuff is for JMS users right? Could this not just sit in the JMS wrappers only then; and avoid it hitting core client code?

Reply:
ClientSession is the only shared object on Every ActiveMQMessage

The other possibility is to revert to the previous change, where you had a boolean on every ActiveMQMessage.

Reply:
Eg the jms connection or jms session

Reply:
Could the jms session not be the thing referenced in the message, instead of the core session. And then this would allow the custom jms bit to live there, and youd still have access to the core session (though probably better to delegate through)

Reply:
Would need a bigger refactoring.  

With the extra code the regular case would be kept untouched.

Reply:
How about using the ConnectionFactoryOptions to hold this, as this is a product of JMS only?

Reply:
I checked your branch, and the method  boolean isEnable1xPrefixes(), doesnt seem to be used anymore at all in your new version so could this just be removed now?

Reply:
@michaelandrepearce I will remove it.. all usage through extensions